addons/sale_project/views/project_task_views.xml

```diff
@@ -100,4 +100,32 @@
         </field>
     </record>
 
+    <record id="project_milestone_view_form" model="ir.ui.view">
+        <field name="name">project.milestone.view.form.inherit</field>
+        <field name="model">project.milestone</field>
+        <field name="inherit_id" ref="project.project_milestone_view_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//group[@name='main_details']" position="after">
+                <group t-if="record.allow_billable.raw_value">
```

Comment:
Is it worked this in the form view? 🤔 
If not then we could do this:
```suggestion
                <group attrs="{'invisible': [('allow_billable', '=', False)]">
```

Reply:
It indeed wasn't working, fixed